In [1]:
## This is a example of how you will be doing projects in python.
### Use markdown cells -OR- comments in cells to place AT LEAST the question number you are working on. 

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
3.	Import the required packages

SyntaxError: invalid syntax (626174333.py, line 1)

In [4]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns

4. Using the pandas `read_csv()` method, read the GDP dataset into your notebook as a DataFrame called `gdp_df`. Take a look at the first few and last few rows to familiarize yourself with what is contained in this dataset.

In [5]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv')

In [6]:
gdp_df.head()

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2023  1992.424394              NaN
1     Afghanistan  2022  1981.710168              NaN
2     Afghanistan  2021  2144.166570              NaN
3     Afghanistan  2020  2769.685745              NaN
4     Afghanistan  2019  2927.245144              NaN

In [7]:
gdp_df.info

<bound method DataFrame.info of      Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2023  1992.424394              NaN
1        Afghanistan  2022  1981.710168              NaN
2        Afghanistan  2021  2144.166570              NaN
3        Afghanistan  2020  2769.685745              NaN
4        Afghanistan  2019  2927.245144              NaN
...              ...   ...          ...              ...
8214        Zimbabwe  1994  3965.730986              NaN
8215        Zimbabwe  1993  3634.750494              NaN
8216        Zimbabwe  1992  3649.891947              NaN
8217        Zimbabwe  1991  4126.405247              NaN
8218        Zimbabwe  1990  4013.299059              NaN

[8219 rows x 4 columns]>

In [ ]:
5. How many rows and columns are in `gdp_df`? What are the data types of each column?

8219 rows, 4 columns

6. Drop the `Value Footnotes` column and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.

In [14]:
gdp_df.rename(columns={'Country or Area': 'Country', 'Year': 'Year', 'Value': 'GDP_Per_Capita'})


Country  Year  GDP_Per_Capita  Value Footnotes
0     Afghanistan  2023     1992.424394              NaN
1     Afghanistan  2022     1981.710168              NaN
2     Afghanistan  2021     2144.166570              NaN
3     Afghanistan  2020     2769.685745              NaN
4     Afghanistan  2019     2927.245144              NaN
...           ...   ...             ...              ...
8214     Zimbabwe  1994     3965.730986              NaN
8215     Zimbabwe  1993     3634.750494              NaN
8216     Zimbabwe  1992     3649.891947              NaN
8217     Zimbabwe  1991     4126.405247              NaN
8218     Zimbabwe  1990     4013.299059              NaN

[8219 rows x 4 columns]

KeyError: "['Value Footnotes'] not found in axis"